modules/migration-configuring-cors-3.adoc

```diff
@@ -0,0 +1,47 @@
+// Module included in the following assemblies:
+//
+// migration/migrating_openshift_3_to_4/migrating-openshift-3-to-4.adoc
+[id='migration-configuring-cors-3_{context}']
+= Configuring cross-origin resource sharing on {product-title} 3 clusters
+
+To access the API server of an {product-title} 3 cluster from a web application (in this case, the Cluster Application Migration tool) on a different host, you must configure cross-origin resource sharing by adding the CAM host name to the master configuration file.
+
+.Procedure
+
+. Log in to the {product-title} 3 cluster.
+. Add the CAM host name to the `corsAllowedOrigins` stanza in the *_/etc/origin/master/master-config.yaml_* configuration file:
++
+----
+corsAllowedOrigins:
+- (?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z)
```

Comment:
I saw some folks very confused by this, thinking literally `(?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z)` is the string that they should use.

I think prior to "Configuring cross-origin resource sharing on OpenShift Container Platform 3 clusters", we need a section that explains how to get the UI's Origin, and we need to make it clear that the docs will continue with `(?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z)` as an example Origin, but this value will be different for every user.

```md
You enable communication with the CAM by configuring cross-origin resource sharing on OpenShift Container Platform 3 and 4 (if the OpenShift Container Platform 4 cluster is not hosting the CAM).

# Obtaining the UI's Origin
First, you must obtain the UI's Origin value to be whitelisted across all of the clusters involved in a migration.

1) You must log into the cluster that is hosting the web UI.
2) The following command will export the encoded `corsAllowedOrigin` for your cluster: `oc get -n openshift-migration route/migration -o go-template='(?i}//{{ .spec.host }}(:|\z){{ println }}' | sed 's,\.,\\.,g'`
3) This output string should be used where an allowed origin must be whitelisted. As an example, we'll use `(?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z)`, but this will differ for your cluster.

```

Reply:
I've added the steps for obtaining the CAM host URL and marked `(?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z)` with a callout saying that it needs to be replaced with the user's CAM host URL.